File: crates/enforcer-memory/tests/fixtures/memory/lang_go/widget_test.go
package widget

import "testing"

func TestNewWidget(t *testing.T) {
	w := NewWidget("foo")
	if w.Name != "foo" {
		t.Fatal("unexpected name")
	}
}

func helperNotATest() {
	// Not a Test-prefixed name, must not be classified as a Test symbol.
}

File: crates/enforcer-memory/tests/fixtures/memory/lang_c/widget_test.c
#include "widget.h"

void test_widget_new_sets_id() {
    struct Widget* w = widget_new("foo");
}

void helper_not_a_test() {
    /* Not a test_-prefixed name, must not be classified as a Test symbol
       when the file itself is not on a *_test.c path -- but this file IS
       named widget_test.c, so the file-level signal promotes it anyway. */
}
